core/sql/optimizer/EncodedKeyValue.cpp

```diff
@@ -69,7 +69,14 @@ NAString * getMinMaxValue(TrafDesc * column,
   
   Lng32 buflen = type->getTotalSize();
   char * buf = new char[buflen]; // deleted at the end of this method
-
+  if (type->supportsSQLnullPhysical())
+  {
+      Lng32 nullHdrSize = type->getSQLnullHdrSize();
+      for(int i = 0; i < nullHdrSize; i++)
```

Comment:
Prior to this change, the nullIndictor will be filled by 0 by accident, since the minimum value is 0. So it will fill in N zeros into the buffer starting from the null header.
I don't know if it is required for other type. This is a good question, let me check other type as well.
But in fact, this change is not necessary. Since the buf is never used in this function.

I will rework on this.